src/mon/FSCommands.cc

```diff
@@ -1526,6 +1514,13 @@ int FileSystemCommandHandler::_check_pool(
 
   const string& pool_name = osd_map.get_pool_name(pool_id);
 
+  if (!allow_overlay && pool->application_metadata.count(pg_pool_t::APPLICATION_NAME_CEPHFS) != 0) {
+    auto kv_pair = pool->application_metadata.find(pg_pool_t::APPLICATION_NAME_CEPHFS)->second.begin();
+    *ss << "RADOS pool '" << pool_name << "' is already used by filesystem '" << kv_pair->second << "'";
```

Comment:
you should use structured bindings when getting access to a pair object, since they are more descriptive:

    auto& [app_name, metadata] = pool->application_metadata.find(pg_pool_t::APPLICATION_NAME_CEPHFS);
    auto& [key, value] = metadata.begin();

    *ss << "RADOS pool '" << pool_name << "' is already used by filesystem '" << value << "'";

You should also lookup the specific key rather than assuming `metadata.begin()` will always be the desired key

Reply:
Done